debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/AbstractColumnValue.java

```diff
@@ -36,12 +38,17 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractColumnValue.class);
 
     @Override
-    public LocalDate asLocalDate() {
+    public Object asLocalDate() {
```

Comment:
Why this need to be changed?

Reply:
The came down to the fact that `PgProtoColumnValue` extended `AbstractColumnValue` and it needed to return a `Duration` rather than a `LocalDate`.  In fact, for all the return type changes I had to make to the abstract class and its interface, these were due to keeping the same return types from the prior code this replaced.  Do you have a more preferred solution for this?

Reply:
I still don't quite understand this. Why wasn't this needed before?

Reply:
When `AbstractColumnValue` was first implemented, it was only used by pgoutput and wal2json, of which both returned a `LocalDate` type.  When migrating decodebufs to use this class when implementing `PgProtoColumnValue`, I changed this to account for the fact that `PgProtoReplicationMessage#getValue` returned different types.

In hindsight I guess I could have left this as-is and instead implemented `PgColumnValue#asLocalDate` as follows:
```
    @Override
    public LocalDate asLocalDate() {
        if (value.hasDatumInt32()) {
            return LocalDate.ofEpochDay((long) value.getDatumInt32());
        }

        final String s = asString();
        return s != null ? DateTimeFormat.get().date(s) : null;
    }
```
Is there a preference here?

Reply:
Yes, that one seems definitely nicer. So perhaps most (all?) return type changes in `AbstractColumnValue` could be reverted then?

Reply:
For now I've reverted all but the ones related to `asTime()` and `asLocalTime()` given that I have yet to find a solution that works with PostgreSQL allowing `24:00:00` in `TIME` fields; see commit 4389c16.

Reply:
For now I've reverted all but the ones related to `asTime()` and `asLocalTime()` given that I have yet to find a solution that works with PostgreSQL allowing `24:00:00` in `TIME` fields; see commit 4389c16.